operator/src/main/java/oracle/kubernetes/operator/helpers/BasePodStepContext.java

```diff
@@ -49,6 +61,37 @@ protected V1Container createPrimaryContainer(TuningParameters tuningParameters)
         .securityContext(getServerSpec().getContainerSecurityContext());
   }
 
+  protected V1Volume createEmptyDirVolume(CommonMount cm) {
+    V1EmptyDirVolumeSource emptyDirVolumeSource = new V1EmptyDirVolumeSource();
+    Optional.ofNullable(cm.getMedium()).ifPresent(medium -> emptyDirVolumeSource.medium(medium));
+    Optional.ofNullable(cm.getSizeLimit())
+            .ifPresent(sizeLimit -> emptyDirVolumeSource.sizeLimit(Quantity.fromString(sizeLimit)));
+    return new V1Volume().name(cm.getEmptyDirVolumeName()).emptyDir(emptyDirVolumeSource);
+  }
+
+  protected V1Container createInitContainerForCommonMount(Container container, int index, CommonMount cm) {
+    return new V1Container().name(getName(container, index))
+        .image(container.getImage())
+            .imagePullPolicy(container.getImagePullPolicy())
+            .command(Collections.singletonList(SHELL))
+            .env(createEnv(cm))
+            .args(Arrays.asList("-c", container.getCommand()))
```

Comment:
I suspect more informative error handling would be helpful here.  In detail, I suspect an error here would result in a somewhat obscure failure and error message at the moment.  

So how about invoking a small wrapper script that provides detailed debugging and failure messages? And that uses the same 'utils' helper trace methods as the main scripts (so that the Operator can parse the output and include any errors it scrapes in its error messages)?  You can put the commands in a shell script in the resources directory.

The script could do something like:

```
common-mount-copy.sh
# Init container script for the common mount feature. 
# See 'domain.spec.serverPod.commonMount' for details.
load 'utils.sh'
trace "Common Mount: About to copy common mount container image=$CONTAINER_IMAGE files in COMMON_DIR=$COMMON_DIR TARGET_DIR=xxx"
trace messages that show current 'id', directory content, and file permissions for the COMMON_DIR in the image (there's a one-line utils.sh helper method for this)
trace ERROR message if COMMON_DIR doesn't exist or is empty (and error out)
trace FINE about to exec COPY_COMMAND=$COPY_COMMAND
results=$(exec "$COPY_COMMAND" 2>&1)
trace ERROR message if fails err=$results
trace FINE message on success out=$results
exit non-zero on failure
```

in addition, there should be integration tests for error paths in ^^^ - if the above errors happen in the introspector, then the introspector is expected to fail, and the ^^^ error messages should show automatically show up in the operator log and the domain status